### PR TITLE
Rework Jetty ThreadPool metrics to work with any ThreadPool

### PR DESCRIPTION
**Pull request by izeye**
This PR updates `JettyServerThreadPoolMetrics` to work with any Jetty `ThreadPool`.

Closes gh-911

### CODE REVIEW

File: micrometer-spring-legacy/src/main/java/io/micrometer/spring/autoconfigure/web/jetty/JettyMetricsAutoConfiguration.java [hunk 1]
```diff
@@ -46,11 +53,26 @@
         "javax.servlet.Servlet", "org.eclipse.jetty.server.Server", "org.eclipse.jetty.util.Loader",
         "org.eclipse.jetty.webapp.WebAppContext" })
 @Conditional(JettyMetricsAutoConfiguration.JettyMetricsAutoConfigurationConditionalOnBeans.class)
+@ConditionalOnBean(AnnotationConfigEmbeddedWebApplicationContext.class)
```
Comment (wilkinsona) [inline micrometer-spring-legacy/src/main/java/io/micrometer/spring/autoconfigure/web/jetty/JettyMetricsAutoConfiguration.java]: Sometimes I am surprised by what's available as a bean. This is one of those times. Hopefully we can come up with a more robust way of dealing with this, but I'm not yet sure what it should be. A custom condition might make the intent clearer if nothing else.

**Comment by wilkinsona** [comment]
I've opened https://github.com/spring-projects/spring-boot/issues/14784.

Another, somewhat related, thought just occurred to me about the difference between Tomcat's metrics and Jetty's metrics. Tomcat's metrics are heavily based on JMX. This, I believe, means that they'll work for applications deploy to a Tomcat container and applications that are using embedded Tomcat. I wonder if a similar approach was considered for Jetty and, if it was, why it was rejected.

File: micrometer-spring-legacy/src/main/java/io/micrometer/spring/autoconfigure/web/jetty/JettyMetricsAutoConfiguration.java (hunk 1 above)
Comment (izeye) [reply micrometer-spring-legacy/src/main/java/io/micrometer/spring/autoconfigure/web/jetty/JettyMetricsAutoConfiguration.java]: @wilkinsona Thanks for the feedback! I changed to use a custom condition to check whether it's an embedded web application in a32aa8a.

**Comment by jkschneider** [comment]
@wilkinsona I can't say there was much deliberate thought here. I tend to want to get things from a programmatic interface if possible, resorting to JMX when there is no other option.

**Pull request merged**